### PR TITLE
Check for empty ConcavePolygonShape2D before checking for intersection

### PR DESCRIPTION
**Pull request by madmiraal**
Fixes another issue identified in #47436.

### CODE REVIEW

File: servers/physics_2d/shape_2d_sw.cpp [hunk 1]
```diff
@@ -691,6 +691,9 @@ bool ConcavePolygonShape2DSW::contains_point(const Vector2 &p_point) const {
 }
 
 bool ConcavePolygonShape2DSW::intersect_segment(const Vector2 &p_begin, const Vector2 &p_end, Vector2 &r_point, Vector2 &r_normal) const {
+	if (segments.size() == 0 || points.size() == 0)
+		return false;
```
Comment (pouleyKetchoupp) [inline servers/physics_2d/shape_2d_sw.cpp:695]: Why not raising an error as in the other cases?

Comment (madmiraal) [reply servers/physics_2d/shape_2d_sw.cpp:695]: In the other cases, we're failing and returning a default value. In this case we can legitimately say there won't be an intersection if there are no segments or points.

Comment (pouleyKetchoupp) [reply servers/physics_2d/shape_2d_sw.cpp:695]: Alright, makes sense. It just needs brackets because of the coding style on master.

Comment (madmiraal) [reply servers/physics_2d/shape_2d_sw.cpp]: Brackets added.

**Review by pouleyKetchoupp — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**